src/server/game/Instances/InstanceScript.cpp

```diff
@@ -535,99 +535,87 @@ void InstanceScript::DoRespawnGameObject(ObjectGuid guid, uint32 timeToDespawn /
         TC_LOG_DEBUG("scripts", "InstanceScript: DoRespawnGameObject failed");
 }
 
-void InstanceScript::DoUpdateWorldState(uint32 uiStateId, uint32 uiStateData)
+void InstanceScript::DoOnPlayers(std::function<void(Player*)>&& function)
 {
-    Map::PlayerList const& lPlayers = instance->GetPlayers();
+    Map::PlayerList const& plrList = instance->GetPlayers();
+
+    if (!plrList.isEmpty())
+        for (Map::PlayerList::const_iterator i = plrList.begin(); i != plrList.end(); ++i)
+            if (Player* player = i->GetSource())
+                function(player);
+}
 
-    if (!lPlayers.isEmpty())
+void InstanceScript::DoUpdateWorldState(uint32 uiStateId, uint32 uiStateData)
+{
+    DoOnPlayers([uiStateId, uiStateData](Player* player)
     {
-        for (Map::PlayerList::const_iterator itr = lPlayers.begin(); itr != lPlayers.end(); ++itr)
-            if (Player* player = itr->GetSource())
-                player->SendUpdateWorldState(uiStateId, uiStateData);
-    }
-    else
-        TC_LOG_DEBUG("scripts", "DoUpdateWorldState attempt send data but no players in map.");
+        player->SendUpdateWorldState(uiStateId, uiStateData);
+    });
 }
 
 // Send Notify to all players in instance
 void InstanceScript::DoSendNotifyToInstance(char const* format, ...)
 {
-    InstanceMap::PlayerList const& players = instance->GetPlayers();
+    va_list ap;
+    va_start(ap, format);
+    char buff[1024];
+    vsnprintf(buff, 1024, format, ap);
+    va_end(ap);
```

Comment:
No.

Reply:
MonkaS

Reply:
This is a refactor of existing method already doing thing, don't complain, its not new code